optaplanner-examples/src/test/java/org/optaplanner/examples/nqueens/solver/score/NQueensConstraintProviderTest.java

```diff
@@ -129,9 +129,20 @@ private static NQueens readSolution(String resource) throws IOException {
     }
 
     @Test
-    public void constraintProviderIntegrationTest() throws IOException {
+    public void constraintProviderSolutionBasedIntegrationTest() throws IOException {
         constraintVerifier.verifyThat(constraintProvider)
                 .given(readSolution("256queensScore-30.xml"))
                 .scores(SimpleScore.of(-30));
     }
+
+    @Test
+    public void constraintProviderFactBasedIntegrationTest() {
```

Comment:
(soft) 0) do we need the prefix "constraitnProvider"? It in the class ...ConstraintProviderTest
1) What method are we testing? Answer: "given". I 'd start with that.
2) I am not a fan of having the word "test" in methods that already have @Test annotation.
3) Today I understand why this is an "integration" test. Because it does more integration than the other unit tests that just test one constraint. In 6 months from now I'll have forgotten that. Can we naming it something like "multipleConstraints"?

Therefore, these method name suggestions for these 2 tests:
givenSolutionMultipleConstraints
givenFactsMultipleConstraints